src/core/simulation.jl

```diff
@@ -74,21 +76,35 @@ get_sequence(s::Simulation) = s.sequence
 get_steps(s::Simulation) = s.steps
 get_date_range(s::Simulation) = s.internal.date_range
 get_stage(s::Simulation, name::String) = get(s.stages, name, nothing)
+get_stage_interval(s::Simulation, name::String) = get_stage_interval(s.sequence, name)
 get_stage(s::Simulation, number::Int) = get(s.stages, s.sequence.order[number], nothing)
-get_last_stage(s::Simulation) = get_stage(s, s.internal.stages_count)
+get_stages_quantity(s::Simulation) = s.internal.stages_count
 function get_simulation_time(s::Simulation, stage_number::Int)
     return s.internal.date_ref[stage_number]
 end
-get_ini_cond_chronology(s::Simulation, number::Int) =
-    get(s.sequence.ini_cond_chronology, s.sequence.order[number], nothing)
-get_name(s::Simulation, stage::Stage) = get(s.sequence.order, get_number(stage), nothing)
+get_ini_cond_chronology(s::Simulation) = s.sequence.ini_cond_chronology
+get_stage_name(s::Simulation, stage::Stage) = get_stage_name(s.sequence, stage)
+get_name(s::Simulation) = s.name
+get_simulation_folder(s::Simulation) = s.simulation_folder
+get_execution_order(s::Simulation) = s.sequence.execution_order
+get_current_execution_index(s::Simulation) = s.sequence.current_execution_index
+get_stage_cache_definition(s::Simulation, stage::String) =
+    get(s.sequence.cache, stage, nothing)
+get_cache(s::Simulation, key::CacheKey) = get(s.internal.simulation_cache, key, nothing)
```

Comment:
Would it be better to throw an exception here if key is not stored?  It depends on whether the calling code will take some other action in that case.

Reply:
I look into the uses of `get_cache` and check if I should throw an error when not present.